src/handler/global/start.ts

```diff
@@ -39,10 +38,28 @@ export class Start {
             });
             return;
         }
-        settingsRepository.saveSettings(interaction.user.id, apiKey, true);
-        await interaction.reply({
-            content: "Saved! (Only partially implemented)",
-            ephemeral: true,
-        });
+        const user = await getUser(apiKey);
+        if (!isErrorResponse(user.data)) {
+            settingsRepository.saveSettings(
+                interaction.user.id,
+                apiKey,
+                true,
+                user.data.id
+            );
+            await interaction.reply({
+                content: "Saved! (Only partially implemented)",
+                ephemeral: true,
+            });
+        } else if (user.data.code === 1) {
+            await interaction.reply({
```

Comment:
For consistency this should throw a `BotError` or reply with a red coloured embed like the other errors